src/arch/src/aarch64/regs.rs

```diff
@@ -80,17 +80,15 @@ const NR_FP_VREGS: usize = 32;
 // The core register are represented by the user_pt_regs structure. Look for it in
 // arch/arm64/include/uapi/asm/ptrace.h.
 
-// This macro gets the offset of a structure (i.e `str`) member (i.e `field`) without having
-// an instance of that structure.
-// It uses a null pointer to retrieve the offset to the field.
-// Inspired by C solution: `#define offsetof(str, f) ((size_t)(&((str *)0)->f))`.
-// Doing `offset__of!(user_pt_regs, pstate)` in our rust code will trigger the following:
-// unsafe { &(*(0 as *const user_pt_regs)).pstate as *const _ as usize }
-// The dereference expression produces an lvalue, but that lvalue is not actually read from,
-// we're just doing pointer math on it, so in theory, it should be safe.
+// Gets offset of a member (`field`) within a struct (`container`).
```

Comment:
What does test from commit message refers to? You are not modifying any test, right?
Can you please modify the commit messages to include at least why was the change necessary?

Reply:
Added to commit message:

> Fixed offset macro used in Arms tests. Previous implementation
dereferenced a null pointer which is newly an error with the Rust
update.

Reply:
The `offset__of` macro is not used in tests only, it is used in production code: https://github.com/firecracker-microvm/firecracker/blob/047e076c402df9438d41ce6dafa9a30ecab616ab/src/arch/src/aarch64/regs.rs#L202.

Reply:
Ah okay, updated title and message:

> Arm offset macro fix
>
> Fixed offset macro used in Arms code. Previous implementation
dereferenced a null pointer which is newly an error with the Rust
update.